### PR TITLE
YAML Linting & Schema Checks

### PR DESCRIPTION
**Pull request by xenoscr**
# Pull Request Summary
This pull request adds YAML linting and schema check workflows to push and pull requests. It should also address issue #249 
 and #250. There is still some testing that needs to be done with the linting and reviewdog workflow that I was unable to test without performing a pull request. This pull adds the following workflows:

## Push Workflows
- YAML Linting & Validation
  - YAML Lint checks with yamllint via: [ibiqlik/action-yamllint@v3](https://github.com/marketplace/actions/yaml-lint)
  - YAML Schema Validation with pyKwalify via: [cketti/action-pykwalify@v0.3-temp-fix](https://github.com/cketti/action-pykwalify)

## Pull Request Workflows
- YAML Linting & Validations
  - YAML Lint checks with yamllint & reviewdog via: [reviewdog/action-yamllint@v1](https://github.com/reviewdog/action-yamllint) (This has not been tested but it should add comments to pull requests automatically to call out errors that must be fixed)
  - YAML Schema Validation with pyKwalify via: [cketti/action-pykwalify@v0.3-temp-fix](https://github.com/cketti/action-pykwalify)

# Fixes
To support these changes, numerous corrections needed to be made to the existing YAML files. Most notably, to validate the Schema properly with pyKwalify the additional "---" needed to be removed from the end of each yml file. This change required that the workflow for the GitHub pages automation needed to be [updated](https://github.com/xenoscr/LOLBAS-1/blob/9aaa4903e93fec90b1ed889993d6f5454145e5cf/.github/workflows/gh-pages.yml#L19). The change adds the "---" back at the end of each file so that they can be correctly interpreted as Markdown files. The following common errors were corrected:

- Incorrect category values (mostly casing issues)
- Incorrect indentation
- Incorrect new-line characters (needed to run dos2unix on a few files)
- Schema violations
  - Various missing keys and values

I recommend we implement this pull and then look into failing pull requests if checks fail.

### CODE REVIEW

**Comment by xenoscr** [comment]
@wietze I had to fall back to normal yamllint for the pull check. The reviewdog workflow was treating warnings as errors despite the setting change to "error". The current linting and schema checks should cut down on our work as members of the community submit new pull requests.

File: YML-Schema.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            type: str
+            required: true
+            enum: [ADS, AWL Bypass, Compile, Copy, Credentials, Decode, Download, Dump, Encode, Execute, Reconnaissance, UAC Bypass, Upload]
+          "Privileges":
+            type: str
+            required: true
+          "MitreID":
+            type: str
+            required: true
+            pattern: 'T[0-9]{4}'
```
Comment (wietze) [inline YML-Schema.yml:46]: To make this stricter, i think it should be:
```regex
^T[0-9]{4}(\.[0-9]{3})?$
```

File: YML-Schema.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  "Acknowledgement":
+    type: seq
+    required: false
+    sequence:
+      - type: map
+        mapping:
+          "Person":
+            type: str
+          "Handle":
+            type: str
```
Comment (wietze) [inline YML-Schema.yml:108]: Maybe also add
```yaml
pattern: '^@(\w){1,15}$'
```

File: YML-Schema.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    type: seq
+    required: false
+    sequence:
+      - type: map
+        mapping:
+          "IOC":
+            type: str
+          "Sigma":
+            type: str
+            pattern: 'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
```
Comment (wietze) [inline YML-Schema.yml:77]: These should probably all be starting with `^` and ending with `$`

File: .github/workflows/yaml-linting.yml [hunk 4]
```diff
@@ -1,19 +1,35 @@
...
       - name: yaml-lint
         uses: ibiqlik/action-yamllint@v3
         with:
+          no_warnings: true
+          file_or_dir: yml/**/*.yml
           config_file: .github/.yamllint
+      - name: Validate OSBinaries YAML Schema
+        uses: cketti/action-pykwalify@v0.3-temp-fix
+        with:
+          files: yml/OSBinaries/*.yml
```
Comment (wietze) [inline .github/workflows/yaml-linting.yml:19]: is there a reason we're splitting these out? can't we do a single one with `files: yml/*/*.yml`?

**Review by wietze — CHANGES_REQUESTED** [review]
Great work, this is really a step forward. A few minor suggestions from me

File: YML-Schema.yml (hunk 3 above)
Comment (xenoscr) [reply YML-Schema.yml:77]: Implementing with next push.

File: .github/workflows/yaml-linting.yml (hunk 4 above)
Comment (xenoscr) [reply .github/workflows/yaml-linting.yml:19]: My thought was to split it out to make it easier to narrow down where a invalid file could be. We could absolutely use something like yml/**/*.yml (I think that's correct?) instead. If we just want to do one big check, that is OK with me.

File: YML-Schema.yml (hunk 2 above)
Comment (xenoscr) [reply YML-Schema.yml:108]: Had to tweak it a bit more to allow for blanks. Not everyone has a handle or wishes to share a handle.

File: YML-Schema.yml (hunk 3 above)
Comment (xenoscr) [reply YML-Schema.yml]: Had to tweak this regex some more to allow anchor tags in URLs.

File: .github/workflows/yaml-linting.yml (hunk 4 above)
Comment (xenoscr) [reply .github/workflows/yaml-linting.yml:19]: This is all that remains, either way the checks take the same time. Happy to change it.

Comment (wietze) [reply .github/workflows/yaml-linting.yml:19]: That's fair, happy to keep it as you have it

**Review by wietze — APPROVED** [review]
Great work. I'm happy with it 👌

**Pull request merged**